src/native/cipher.rs

```diff
@@ -261,3 +258,305 @@ pub fn decrypt_in<P: PoolParams>(eta:Num<P::Fr>, memo:&[u8], params:&P)->Vec<Opt
         vec![]
     }
 }
+
+/// get encrypted memo chunks with associated decryption keys (chunk: account or note)
+/// returns vector of tupple (index, chunk, key)
+/// indexes are zero-based and enumerated within current memo
+pub fn symcipher_decryption_keys<P: PoolParams>(eta:Num<P::Fr>, mut memo:&[u8], params:&P) -> Option<Vec<(u64, Vec<u8>, Vec<u8>)>> {
+    let num_size = constants::num_size_bits::<P::Fr>()/8;
+    let account_size = constants::account_size_bits::<P::Fr>()/8;
+    let note_size = constants::note_size_bits::<P::Fr>()/8;
+
+    let nozero_items_num = u32::deserialize(&mut memo).ok()? as usize;
+    if nozero_items_num == 0 {
+        return None;
+    }
+
+    let nozero_notes_num = nozero_items_num - 1;
+    let shared_secret_ciphertext_size = nozero_items_num * constants::U256_SIZE + constants::POLY_1305_TAG_SIZE;
+
+    let account_hash = Num::deserialize(&mut memo).ok()?;
+    let note_hashes = buf_take(&mut memo, nozero_notes_num * num_size)?;
+
+    let shared_secret_text = {
```

Comment:
Perhaps it makes sense to move it to a separate function as you did with the decryption of the account and notes.

Reply:
Good idea. Please check it in https://github.com/zkBob/libzeropool/pull/9/commits/a47890b368d10232e90880fab6f6ef89da53eb9d